lib/TWCManager/TWCMaster.py

```diff
@@ -532,6 +533,29 @@ def getMaxAmpsToDivideGreenEnergy(self):
         amps = amps / self.getRealPowerFactor(amps)
         return round(amps, 2)
 
+    def getMaxAmpsToDivideFromGrid(self):
+        # Calculate our current generation and consumption in watts
+        generationW = float(self.getGeneration())
+        consumptionW = float(self.getConsumption())
+
+        currentOffer = min(
+            self.getTotalAmpsInUse(),
+            self.getMaxAmpsToDivideAmongSlaves(),
+        )
+
+        # Calculate what we should max offer to align with max grid energy
+        amps = self.config["config"]["maxAmpsAllowedFromGrid"] + \
+               self.convertWattsToAmps(generationW - consumptionW) + \
```

Comment:
We have a solar production that is connected to a own fuse, Consumption has a smaller fuse. So the solar production is not really help to increase the maxAmpsAllowedFromGrid. I guess this case needs to/could be added as a setting to ignore generationW.

Reply:
It should be controled latelly by the wiringMaxAmpsAllTWCs and wiringMaxAmpsPerTWC, despite of the value given to the maxAmpsToDivideAmongSlaves it's always limited to the wiring limit.

Reply:
But wiringMaxAmpsAllTWCs and wiringMaxAmpsPerTWC are hardcoded, right?

I'd like the charge power to be dynamic based on the other consumers in the house. So if the rest of the house consume 5A out of 16A, the car can still charge with 9A.